python/mlx/optimizers.py

```diff
@@ -423,6 +423,10 @@ def __init__(
         self, learning_rate: float, betas: List[float] = [0.9, 0.999], eps: float = 1e-8
     ):
         super().__init__(learning_rate, betas, eps)
+        if not 0.0 <= learning_rate:
+            raise ValueError(f"Learning rate should be >=0, {self.learning_rate} was provided instead")
```

Comment:
Is it necessary that the learning rate be `>0`. It seems like a convenient way to do maximization would be to allow positive learning rates. Don't think we should throw in that case

Reply:
I have not come across any studies regarding the use of negative learning rates before. When I look at modern frameworks like TensorFlow and PyTorch, they also assume positive learning rates. Of course, as you suggested, we can do it differently, but I believe we can stick with positive learning rates in MLX as other frameworks do. If there are any studies related to this topic, could you please provide the names? I will make the change accordingly

Reply:
Yea might it might be safer indeed for this to only use positive learning rates. For SGD it's pretty basic:

`p + lr * g` vs `p - lr * g` does maximization vs minimization.